draft-ietf-quic-v2.md

```diff
@@ -139,13 +139,25 @@ nonce = 0x141b99c239b03e785d6a2e9f
 
 # Version Negotiation Considerations
 
-QUIC version 2 endpoints SHOULD also support QUIC version 1. Any QUIC endpoint
-that supports multiple versions MUST meet the minimum requirements described in
-{{QUIC-VN}} to prevent version downgrade attacks.
+QUIC version 2 is not intended to deprecate version 1. Endpoints that support
+version 2 might continue support for version 1 to maximize compatibility
+with clients. In particular, HTTP clients often use Alt-Svc {{?RFC7838}} to
+discover QUIC support. As this mechanism does not currently distinguish between
+QUIC versions, HTTP servers that support multiple versions reduce the
+probability of incompatibility and the cost associated with QUIC version
+negotiation or TCP fallback.
+
+Any QUIC endpoint that supports multiple versions MUST meet the minimum
+requirements described in {{QUIC-VN}} to prevent version downgrade attacks.
 
 Note that version 2 meets that document's definition of a compatible version
-with version 1. Therefore, v2-capable servers MUST use compatible version
-negotiation unless they do not support version 1.
+with version 1. Therefore, servers can use compatible negotiation to switch a
+connection between the two versions.
+
+A server might fully support version 2 but merely parse Initial packets from
+version 1, or vice versa. Such servers SHOULD use compatible negotiation instead
+of sending a Version Negotiation packet to initiate incompatible version
+negotiation to avoid the delay penalty of the latter.
```

Comment:
This paragraph is odd: servers won't drop support and maintain minimal initial parsing unless they're doing compatible VN. Perhaps:
```suggestion
If a server supporting version 2 wishes to drop support for version 1, it SHOULD
retain the ability to parse Initial packets from version 1 and support compatible
negotiation from version 1 to version 2; that will avoid the delay caused by
incompatible version negotiation.
```

Reply:
Yes, no one's going to "drop" v1 support. But someone could implement a server from scratch and only do v2. Similarly, the case where a v1 server decides to do the minimum to gracefully handle v2 seems really plausible to me.

There's plenty of text before that that says "no really, don't delete your v1 code base". The intent of this is to (1) raise partial support as a possibility, but (2) say that if you do, you SHOULD use compatible, not rely on incompatible.

Reply:
But why would anyone do partial support but not compatible VN?